flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -600,8 +601,15 @@ private ChangelogSnapshotState completeRestore(
                                     System.currentTimeMillis(),
                                     streamFactory,
                                     CHECKPOINT_OPTIONS),
-                            // TODO: add metadata to log FLINK-23170.
-                            upTo));
+                            upTo);
+
+            // log metadata after materialization is triggered
+            for (InternalKvState<K, ?, ?> changelogState : keyValueStatesByName.values()) {
+                checkState(changelogState instanceof ChangelogState);
+                ((ChangelogState) changelogState).resetWritingMetaFlag();
```

Comment:
The loop is still through `keyValueStatesByName` only which only contains KV states.
What I meant is to loop through `priorityQueueStatesByName` as well which contains PQ states.

nit: checkState is unnecessary because JRE will check this on the next line.